engine/CMakeLists.txt

```diff
@@ -1530,3 +1536,36 @@ ELSE ()
 ENDIF (WIN32 AND NOT UNIX)
 
 INCLUDE(CPack)
+
+IF (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} GREATER 2.10)
```

Comment:
This should be `3.10`, not `2.10`. And `VERSION_GREATER` for comparing versions, rather than simply `GREATER`.

Let's see if those two changes fix it.

Reply:
Unfortunately, I tested a simple if(false) { jhfdskhdsklh } and cmake failed to parse. So this is not a preprocessor. You can't have stuff that's unknown to the cmake version anywhere in the file.